### DISCUSSION TITLE
Support for hot reloading of TLS credentials for exporters?

### DISCUSSION THREAD

**Discussion by evantorrie** [open-telemetry/opentelemetry-java#4720] [Q&A]
Do the SDK exporters (OTLP/Zipkin/Jaeger) support hot-reloading of TLS credentials?

e.g. watching for changes in the files referenced by environment variables OTEL_EXPORTER_OTLP_CERTIFICATE, OTEL_EXPORTER_OTLP_CLIENT_KEY and OTEL_EXPORTER_OTLP_CLIENT_CERTIFICATE and then updating said credentials when the contents of those files change?

**Comment by jkwatson** [comment]
They do not, currently.

**Reply by evantorrie** [reply]
Our deployments tend to have fairly frequently rotated certificates (e.g. every 24 hours), so this is something we would be interested in requesting.  Do you suggest opening an Issue?

**Reply by jkwatson** [reply]
yes, and if this is something important, contributing the implementation will probably make it more likely that it happens.

**Reply by trask** [reply]
(adding link to #4725)
